eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigReplacerProcessor.java

```diff
@@ -23,70 +25,49 @@
-
-        if (!contigPresentInAssemblyReport && !chromosomePresentInAssemblyReport) {
-            throw new IllegalStateException(
-                    "Neither contig '" + subSnpNoHgvs.getContigName() + "' nor chromosome '"
-                            + subSnpNoHgvs.getChromosome()
-                            + "' were found in the assembly report! Is the assembly accession '"
-                            + assemblyAccession + "' correct?");
-        }
-
-        if (chromosomePresentInAssemblyReport
-                && contigPresentInAssemblyReport
-                && !contigSynonyms.equals(chromosomeSynonyms)) {
-            throw new IllegalStateException(
-                    "Contig '" + subSnpNoHgvs.getContigName() + "' and chromosome '" + subSnpNoHgvs.getChromosome()
-                            + "' do not appear in the same line in the assembly report!");
-        }
-
-        if (contigPresentInAssemblyReport) {
-            replaceContigWithGenbankAccession(subSnpNoHgvs, contigSynonyms);
-        } else {
-            replaceChromosomeWithGenbankAccession(subSnpNoHgvs, chromosomeSynonyms);
+        if (isContigPresentInAssemblyReport(subSnpNoHgvs, contigSynonyms)) {
+            if (contigSynonyms.isIdenticalGenBankAndRefSeq()) {
+                subSnpNoHgvs.setContigName(contigSynonyms.getGenBank());
+            }
         }
 
         return subSnpNoHgvs;
     }
 
-    private void replaceContigWithGenbankAccession(SubSnpNoHgvs subSnpNoHgvs, ContigSynonyms contigSynonyms) {
-        if (contigSynonyms.isIdenticalGenBankAndRefSeq() || isGenbank(assemblyAccession)) {
-            subSnpNoHgvs.setContigName(contigSynonyms.getGenBank());
-        } else {
-            // genbank is not identical to refseq and the assembly is not genbank, so
-            // we must keep the original refseq
-        }
-    }
-
-    private void replaceChromosomeWithGenbankAccession(SubSnpNoHgvs subSnpNoHgvs, ContigSynonyms chromosomeSynonyms) {
-        if (chromosomeSynonyms.isIdenticalGenBankAndRefSeq() || isGenbank(assemblyAccession)) {
-            subSnpNoHgvs.setContigName(chromosomeSynonyms.getGenBank());
-            subSnpNoHgvs.setContigStart(subSnpNoHgvs.getChromosomeStart());
-        } else {
-            // genbank is not identical to refseq and the assembly is not genbank, so
-            // we must keep the original refseq, even if the refseq was not found in the assembly report
+    private boolean isContigPresentInAssemblyReport(SubSnpNoHgvs subSnpNoHgvs, ContigSynonyms contigSynonyms) {
+        boolean contigPresentInAssemblyReport = contigSynonyms != null;
+        if (!contigPresentInAssemblyReport) {
+            logger.warn(
+                    "CONTIG NOT FOUND: Contig '" + subSnpNoHgvs.getContigName() +
+                            "' was not found in the assembly report! Is the " + "assembly accession '" +
+                            assemblyAccession + "' correct?");
```

Comment:
I observed that this scenario is already covered by https://github.com/jmmut/eva-accession/blob/feature/several-assembled-molecules/eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/FastaSynonymSequenceReader.java#L39. I feel that this error belongs in FastaSynonymSequenceReader because it is a potentially reusable class and having a guard against Contig not found in the assembly report is better. @jmmut's opinion is that for this particular error, the ContigReplacer is a better option. @cyenyxe @andresfsilva Any thoughts?

Reply:
I'm not sure I understand your point. My point is that the default behaviour (in FastaSynonymSequenceReader) should be throwing an exception if the contig is not found, but for the case of the contig replacer we know this is not an error so we can catch the exception and only log it. Other usages of the FastaSynonymSequenceReader can rely on the exception-throwing behaviour.

Of course what I suggest does not apply if we decide that the general case should expect missing contigs in the assembly report, which is kind of true if we take into account all the strange contigs. However, I don't think this should be the general case because we will accept INSDC sequences only in most parts of EVA.